packages/amplify_api/android/src/main/kotlin/com/amazonaws/amplify/amplify_api/FlutterApiRequest.kt

```diff
@@ -15,132 +15,123 @@
 import com.amplifyframework.api.ApiException
 import com.amplifyframework.api.rest.RestOptions
-import io.flutter.plugin.common.MethodChannel
 
+object FlutterApiRequest {
+    private const val REST_OPTIONS_KEY = "restOptions"
+    private const val CANCEL_TOKEN_KEY = "cancelToken"
 
-class FlutterApiRequest {
-    companion object {
+    private const val API_NAME_KEY = "apiName"
+    private const val PATH_KEY = "path"
+    private const val BODY_KEY = "body"
+    private const val QUERY_PARAM_KEY = "queryParameters"
+    private const val HEADERS_KEY = "headers"
 
-        private val REST_OPTIONS_KEY = "restOptions"
-        private val CANCEL_TOKEN_KEY = "cancelToken"
-
-        private val API_NAME_KEY = "apiName"
-        private val PATH_KEY = "path"
-        private val BODY_KEY = "body"
-        private val QUERY_PARAM_KEY = "queryParameters"
-        private val HEADERS_KEY = "headers"
-
-        // ====== Rest API ======
-        fun getCancelToken(request: Map<String, Any>): String {
-            try {
-                return request[CANCEL_TOKEN_KEY] as String
-            } catch (cause: Exception) {
-                throw AmplifyException(
-                        "The cancelToken request argument was not passed as a String",
-                        cause,
-                        "The request should include the cancelToken as a String")
-            }
+    // ====== Rest API ======
+    fun getCancelToken(request: Map<String, Any>): String {
+        try {
             return request[CANCEL_TOKEN_KEY] as String
+        } catch (cause: Exception) {
+            throw AmplifyException(
+                "The cancelToken request argument was not passed as a String",
+                cause,
+                "The request should include the cancelToken as a String"
+            )
         }
+    }
 
-        fun getApiPath(request: Map<String, Any>): String? {
-            try {
-                val restOptionsMap = request[REST_OPTIONS_KEY] as Map<String, Any>
-                return restOptionsMap[API_NAME_KEY] as String?
-            } catch (cause: Exception) {
-                throw AmplifyException(
-                        "The apiPath request argument was not passed as a String",
-                        cause,
-                        "The request should include the apiPath as a String")
-            }
-            return request[CANCEL_TOKEN_KEY] as String
+    private fun getApiName(requestMap: Map<String, Any>?): String? {
+        try {
+            return requestMap?.get(API_NAME_KEY) as String?
+        } catch (cause: Exception) {
+            throw AmplifyException(
+                "The apiName request argument was not passed as a String",
+                cause,
+                "The request should include the apiName as a String"
+            )
         }
+    }
 
-        fun getRestOptions(request: Map<String, Any>): RestOptions {
+    fun getRestApiName(request: Map<String, Any>): String? {
```

Comment:
Are these additions a direct result of lint results, or are they more general refactoring? If the latter, then they might be better in a different PR since this one is already substantial.

Reply:
This one is a bit of a catch-22. Fixing the lint errors actually caused some tests to start failing. It turned out that there was a latent logical error which was being masked. There is only one real change in this file to fix that, but it looks like a lot more.